test/unit-tests/common/test_image_star.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""
+    :copyright: Copyright 2017-2019 by the contributors (see AUTHORS file).
+    :license: BSD-2-Clause, see LICENSE for details.
+"""
+
+from sphinxcontrib.confluencebuilder.state import ConfluenceState
+from sphinxcontrib_confluencebuilder_util import ConfluenceTestUtil as _
+from sphinxcontrib.confluencebuilder.builder import ConfluenceBuilder
+import os
+import re
+import unittest
+import tempfile
+import shutil
+
+
+class TestImageStar(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.extension_pattern = '@REPLACE_EXTENSION@'
+        cls.test_dir = os.path.dirname(os.path.realpath(__file__))
+        cls.image_file = os.path.join(cls.test_dir, 'assets', 'image01.png')
+
+    def build_document_given_dataset(self, dataset):
+        config = _.prepareConfiguration()
+        doc_dir, doctree_dir = _.prepareDirectories('image-star')
+        _.buildSphinx(dataset, doc_dir, doctree_dir, config)
+        return doc_dir
+
+    def write_expected_file(self, pattern_file, outfile, actual_extension):
+        ''' writes the content of pattern_file into outfile replacing
+        the image pattern by the actual extension
+        '''
+        with open(pattern_file) as inp, open(outfile, 'w') as out:
+            content = ''.join(inp.readlines())
+            out.write(content.replace(self.extension_pattern, actual_extension))
+
+    def test_all_supported_extensions(self):
+        ''' For every supported extension by the Builder
+        a temporary sourcesdir is created with the original index.rst
+        and a link to the original image file. The link name will end
+        with the tested extension. Then the document is built.
+        The expected file is created for every image type just replacing
+        a pattern (@REPLACE_EXTENSION@) in one input file
+        '''
+        dataset = os.path.join(self.test_dir, 'dataset-image-star')
+        image_basename = os.path.basename(self.image_file).split('.')[0]
+        for image_type in ConfluenceBuilder.supported_image_types:
+            specifig_dataset = tempfile.mkdtemp()
+            image_extension = re.sub(r'.*/', '', image_type)
+            os.symlink(os.path.join(dataset, 'index.rst'),
```

Comment:
Curious -- will `os.symlink` work in Windows (without additional system platform modifications)?

Reply:
Good point. I didn't know tests must be cross-platform. I learned os.symlink does not work in Windows for all python versions. Eventually, I am copying the file with shutil module and ensuring the created directory is deleted at the end of the test regardless of the result.